src/services/url.ts

```diff
@@ -10,54 +11,74 @@ export const STOREFRONT_APP_ID = 'GroupBy StoreFront';
 class UrlService extends BaseService<UrlService.Options> {
 
   beautifier: UrlBeautifier;
+  urlState: UrlService.UrlStateFunctions = {
+    search: UrlService.searchUrlState,
+    details: UrlService.detailsUrlState,
+    navigation: UrlService.navigationUrlState
+  };
+  mergeState: UrlService.MergeFunctions = {
+    search: UrlService.mergeSearchState,
+    details: UrlService.mergeDetailsState,
+    navigation: UrlService.mergeNavigationState
+  };
+
+  constructor(app: StoreFront, opts: any) {
+    super(app, opts);
+    this.beautifier = new UrlBeautifier(this.opts.routes, this.opts.beautifier, this.app.config);
```

Comment:
I think we should be able to do this inline with the declaration rather than in the `constructor()`